src/twisted/internet/protocol.py

```diff
@@ -527,17 +527,17 @@ def connectionMade(self):
 @implementer(interfaces.IProtocol, interfaces.ILoggingContext)
 class Protocol(BaseProtocol):
     """
-    This is the base class for streaming connection-oriented protocols.
+    Protocol is a legacy interface for inheritance-based code sharing
+    for streaming connection-oriented protocols.
 
-    If you are going to write a new connection-oriented protocol for Twisted,
-    start here.  Any protocol implementation, either client or server, should
-    be a subclass of this class.
+    New code should prefer the composition-based interface L{interfaces.IProtocol},
+    for either client or server side connections.
 
     The API is quite simple.  Implement L{dataReceived} to handle both
     event-based and synchronous input; output can be sent through the
-    'transport' attribute, which is to be an instance that implements
-    L{twisted.internet.interfaces.ITransport}.  Override C{connectionLost} to be
-    notified when the connection ends.
+    L{BaseProtocol.transport} attribute, which is to be an instance that implements
```

Comment:
in the parlance of zope.interface, the instance *provides* `ITransport`.  It is the type of the instance which *implements* it.